lib/pbench/server/filetree.py

```diff
@@ -336,72 +336,72 @@ def unpack(self, incoming: Path, results: Path):
             the tarball.
         """
         unpacked = incoming / f"{self.name}.unpack"
-        unpacked.mkdir(parents=True)
+        # unpacked.mkdir(parents=True)
 
         try:
             process = subprocess.run(
                 [
                     "tar",
-                    "-xf",
+                    "-x",
                     "--no-same-owner",
                     "--delay-directory-restore",
                     "--force-local",
-                    self.tarball_path,
+                    "--file=" + str(self.tarball_path),
                 ],
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballUnpackError(self.name, str(exc)) from exc
+            raise TarballUnpackError(self.tarball_path, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballUnpackError(
-                    self.name,
+                    self.tarball_path,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
+        find_command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
         try:
             process = subprocess.run(
-                command.split(),
+                find_command.split(),
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballModeChangeError(self.name, str(exc)) from exc
+            raise TarballModeChangeError(self.tarball_path, str(exc)) from exc
```

Comment:
I question using `tarball_path` here, since we're no longer operating on the tarball, but on the unpacked directory. In fact, since the client shouldn't *care* where the tarball itself lives, I think all of these should probably go back to `self.name`.

(Maybe we ought to be using `resource_id` since `name` isn't guaranteed unique; but `name` is likely to be more meaningful in a log. Maybe. At least it makes the log easier to read? 😆 )